botorch_community/acquisition/bll_thompson_sampling.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import annotations
+
+import numpy as np
+import scipy
+
+import torch
+
+from botorch.logging import logger
+
+from botorch_community.models.blls import AbstractBLLModel
+from torch.func import grad
+
+
+class BLLMaxPosteriorSampling:
```

Comment:
There is a [`SamplingStrategy`](https://github.com/pytorch/botorch/blob/main/botorch/generation/sampling.py#L39) interface here that that defines the API for this (and subclasses `torch.nn.Module`) but doesn't support not providing an `X_cand`. I assume this is why you didn't subclass from that? 

I don't think it's necessary to use this if it would require changes to `SamplingStrategy` but maybe leave a comment here that we may want to consider doing that in the future.